pyads/pyads_ex.py

```diff
@@ -835,12 +835,9 @@ def adsSyncReadReqEx2(
     index_group_c = ctypes.c_ulong(index_group)
     index_offset_c = ctypes.c_ulong(index_offset)
 
-    if type_is_string(data_type):
-        data = (STRING_BUFFER * PLCTYPE_STRING)()
-    elif type_is_wstring(data_type):
-        data = (STRING_BUFFER * ctypes.c_uint8)()
-    else:
-        data = data_type()
+    # Strings were handled specifically before, but their sizes are contained and we
+    # can proceed as normal:
+    data = data_type()
```

Comment:
Hm, I just realized this will break a usage for unknown string sizes, like:
```
plc.read_by_name("VAR", data_type=PLCTYPE_STRING)
```

Reply:
Resolved in 0a259d8e , by assuming a singular character actually corresponds to an unknown size.